File: codechef/whats-the-rank/whats-the-rank.py
n = int(input())
pos = []
for _ in range(n):
    a = int(input())
    v = len(pos)
    for i in range(len(pos)):
        if a>pos[i]:
            v=i+1
            pos.insert(i, a)
            break
    if v==0:
        pos.append(a)
        print(len(pos))
    elif v==len(pos):
        pos.append(a)
        print(len(pos))
    else:
        print(v)

File: codechef/whats-the-rank/whats-the-rank.cpp
#include <ext/pb_ds/assoc_container.hpp> // Common file
#include <ext/pb_ds/tree_policy.hpp> // Including tree_order_statistics_node_update
#include <bits/stdc++.h>
#define ll long long
using namespace std;
using namespace __gnu_pbds;

typedef tree <int, null_type, less<int>, rb_tree_tag, tree_order_statistics_node_update> ordered_set;

int main(int argc, char **argv)
{
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    ordered_set pos;
    int n, v;
    cin >> n;
    for(ll i=0; i<n; ++i)
    {
        cin >> v;
        pos.insert(v);
        cout << pos.size() - pos.order_of_key(v) << endl;
    }
    return 0;
}